homework3/cookie-clicker-lab4-7/src/main/java/edu/csula/web/AdminGeneratorsServlet.java

```diff
@@ -0,0 +1,92 @@
+package edu.csula.web;
+
+import edu.csula.models.Generator;
+import edu.csula.storage.GeneratorsDAO;
+import edu.csula.storage.UsersDAO;
+import edu.csula.storage.servlet.GeneratorsDAOImpl;
+import edu.csula.storage.servlet.UsersDAOImpl;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collection;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/generators")
+public class AdminGeneratorsServlet extends HttpServlet {
+	@Override
+	public void doGet( HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		response.setContentType("text/html");
+		PrintWriter out = response.getWriter();
+		// TODO: render the generators page HTML
+
+        UsersDAO daosession=new UsersDAOImpl(request.getSession());
+        if(daosession != null) {
```

Comment:
This logic of checking if user is login or not is wrong. Should be `daosession.getAuthenticateUser().isPresent()`